pkgs/development/compilers/ghc/head.nix

```diff
@@ -12,7 +12,7 @@ let
 
   commonBuildInputs = [ ghc perl autoconf automake happy alex python3 ];
 
-  version = "8.1.20170106";
+  version = "2017-01-06";
```

Comment:
Addressing @peti's objection: the git-generated version is `ghc-8.3-start-421-g43a31683ac`; the revision `b4f2afe70ddbd0576b4eba3f82ba1ddc52e9b3bd` doesn't have a tag, so I assume this revision was the HEAD commit of the time. And I see that "8.1.20170106" is a convention specific to ghc HEAD/unstable package rather than nix in general.